### ISSUE TITLE
TVVLCKit on Apple TV

### ISSUE THREAD

**Issue by sssionggg** [siong1987/siongTime-tvos#1]
If we can get TVVLCKit on Apple TV, it will help a lot on playing the video/audio stream that we want without worrying about the file format.

**Comment by falk** [comment]
Perhaps ffmpeg could be run on the appletv instead of the web backend:  https://github.com/kewlbear/FFmpeg-iOS-build-script

**Comment by sssionggg** [comment]
@falk yes, that's exactly what I am looking at.  it will be cool if someone can get it up and running on the AppleTV.

**Comment by falk** [comment]
There is also https://gist.github.com/wtsnz/ade50e67ae6b006c1db3

**Comment by sssionggg** [comment]
@falk i tried the above script too.  the TVVLCKit didn't work as expected.

**Comment by falk** [comment]
I think you can use buildMobileVLCKit.sh from the official repo now: https://code.videolan.org/videolan/VLCKit/tree/iOS-2.7

There is also the precompiled lib from this post: https://forum.videolan.org/viewtopic.php?f=36&t=128638&start=40#p433969

**Comment by sssionggg** [comment]
@falk awesome.  you should try it out this weekend and see whether it works :)

**Comment by sssionggg** [comment]
@falk if you successfully build it, let me know.  i can figure it out from there.

**Comment by falk** [comment]
I will try to get it up and running in between work and my thesis.

**Comment by falk** [comment]
Also found https://github.com/sylverb/NAStify which seems to have a working tvos app using tvvlckit

**Comment by sssionggg** [comment]
@falk awesome thanks!  let me know once you have it up and running.

**Comment by falk** [comment]
I have not had the time to look at it, but have another idea: Present the streams from a dlna server like https://github.com/Janhouse/torrent-stream-dlna and let localcast/infuse/some other dlna player browse the folders and play the stream.

**Comment by sssionggg** [comment]
@falk the goal is to get rid of server dependency.  i don't think that it will be ideal.

**Comment by falk** [comment]
Here is as vlc example project: https://forum.videolan.org/viewtopic.php?f=36&t=129799

**Comment by sssionggg** [comment]
@falk have you tried to get it work?

**Comment by falk** [comment]
Have not had the time. But just found https://github.com/pepibumur/PopcornTimeTV

**Comment by sssionggg** [comment]
@falk that is cool.  does it work?

**Comment by falk** [comment]
Yes, atm it is hard coded to play a torrent, but I think it can be combined with the interface you have already made.

**Comment by sssionggg** [comment]
@falk also, it is using the default `AVPlayerViewController`.  it won't work with most video formats for BT tho.

**Comment by falk** [comment]
It seems they are going to use vlckit.
